tasks/test.py

```diff
@@ -33,26 +34,55 @@ def ensure_bytes(s):
     return s
 
 
+@contextmanager
+def environ(env):
+    original_environ = os.environ.copy()
+    os.environ.update(env)
+    yield
+    for var in env.keys():
+        if var in original_environ:
+            os.environ[var] = original_environ[var]
+        else:
+            os.environ.pop(var)
+
+
 TOOL_LIST = [
     'github.com/client9/misspell/cmd/misspell',
     'github.com/frapposelli/wwhrd',
     'github.com/fzipp/gocyclo',
     'github.com/go-enry/go-license-detector/v4/cmd/license-detector',
-    'github.com/golangci/golangci-lint/cmd/golangci-lint',
     'github.com/gordonklaus/ineffassign',
     'github.com/goware/modvendor',
     'github.com/mgechev/revive',
+    'github.com/stormcat24/protodep',
+    'golang.org/x/lint/golint',
     'gotest.tools/gotestsum',
     'honnef.co/go/tools/cmd/staticcheck',
 ]
 
+TOOL_LIST_GET = [
+    'github.com/golangci/golangci-lint/cmd/golangci-lint@v1.40.1',
+    'github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway@v1.12.2',
+    'github.com/golang/protobuf/protoc-gen-go@v1.3.2',
+    'github.com/golang/mock/mockgen@v1.5.0',
+]
+
 
 @task
 def install_tools(ctx):
     """Install all Go tools for testing."""
-    with ctx.cd("internal/tools"):
-        for tool in TOOL_LIST:
-            ctx.run("go install {}".format(tool))
+    with environ({'GO111MODULE': 'on'}):
+        with ctx.cd("internal/tools"):
+            for tool in TOOL_LIST:
+                ctx.run("go install {}".format(tool))
+
+            for tool in TOOL_LIST_GET:
+                pkg = tool.split('@')[0]
+                ctx.run("go get {}".format(tool))
```

Comment:
`go get --mod=readonly <tool>` so you don't need the tidy below.

PS: I also could be wrong, but I think you just aren't supposed to use `go install` nowadays with go modules but maybe Pablo can correct me. I think [there's fixes for 1.16](https://github.com/golang/go/issues/40276) but I haven't followed along the changes recently.

Reply:
Oh yes, way better.  Indeed this can be done with go 1.16 simply as `go install some/pkg/foo@vX.Y.Z` we can hopefully clean this up surely after.